src/ApiService/ApiService/onefuzzlib/orm/Orm.cs

```diff
@@ -0,0 +1,265 @@
+        var parameterExpressions =
+            constructorInfo.GetParameters().Select((parameterInfo, i) =>
+            {
+                var ithIndex = Expression.Constant(i);
+                var ithParameter = Expression.ArrayIndex(constructorParameters, ithIndex);
+                var unboxedIthParameter = Expression.Convert(ithParameter, parameterInfo.ParameterType);
+                return unboxedIthParameter;
+
+            }).ToArray();
+
+        NewExpression constructorCall = Expression.New(constructorInfo, parameterExpressions);
+
+        Func<object[], object> ctor = Expression.Lambda<Func<object[], object>>(constructorCall, constructorParameters).Compile();
+        return ctor;
+    }
+
+    private EntityInfo GetEntityInfo<T>()
+    {
+        return _cache.GetOrAdd(typeof(T), type =>
+        {
+            var constructor = type.GetConstructors()[0];
+            var parameterInfos = constructor.GetParameters();
+            var parameters =
+            parameterInfos.Select(f =>
+            {
+                var isRowkey = f.GetCustomAttribute(typeof(RowKeyAttribute)) != null;
+                var isPartitionkey = f.GetCustomAttribute(typeof(PartitionKeyAttribute)) != null;
+
+
+                var (dbName, kind) =
+                isRowkey
+                    ? ("RowKey", EntityPropertyKind.RowKey)
+                    : isPartitionkey
+                        ? ("PartitionKey", EntityPropertyKind.PartitionKey)
+                        : (// JsonPropertyNameAttribute can only be applied to properties
+                            typeof(T).GetProperty(f.Name)?.GetCustomAttribute<JsonPropertyNameAttribute>()?.Name
+                                ?? CaseConverter.PascalToSnake(f.Name),
+                            EntityPropertyKind.Column
+                        );
+                if (f.Name == null)
+                {
+                    throw new Exception();
+                }
+
+                if (f.ParameterType == null)
+                {
+                    throw new Exception();
+                }
+
+                return new EntityProperty(f.Name, dbName, f.ParameterType, kind);
+            }).ToArray();
+
+            return new EntityInfo(typeof(T), parameters, BuildConstructerFrom(constructor));
+        });
+    }
+
+    public TableEntity ToTableEntity<T>(T typedEntity)
+    {
+        if (typedEntity == null)
+        {
+            throw new NullReferenceException();
+        }
+        var type = typeof(T)!;
+        if (type is null)
+        {
+            throw new NullReferenceException();
+        }
+        var tableEntity = new TableEntity();
+        var entityInfo = GetEntityInfo<T>();
+        foreach (var prop in entityInfo.properties)
+        {
+            var value = entityInfo.type.GetProperty(prop.name)?.GetValue(typedEntity);
+            if (prop.type == typeof(Guid) || prop.type == typeof(Guid?))
+            {
+                tableEntity.Add(prop.dbName, value?.ToString());
```

Comment:
does dbName means name inside of the database ? Like a ColumnName ? 

Also should this account for Property Name attribute to allow us rename properties ?

Reply:
yep columnName is a better name. 
Yes we support renaming, see `GetEntityInfo`